src/Cli/func/Helpers/GlobalCoreToolsSettings.cs

```diff
@@ -99,6 +99,13 @@ public static void Init(ISecretsManager secretsManager, string[] args)
             {
                 _currentWorkerRuntime = WorkerRuntime.None;
             }
+
+            if (_currentWorkerRuntime == WorkerRuntime.Java || args.Any(arg => string.Equals(arg, WorkerRuntime.Java.ToString(), StringComparison.OrdinalIgnoreCase)))
```

Comment:
Is this the right place for this check? It seems like this impacts all actions, perhaps not when intended. Even `func host start --java` would get this warning, which seems a little overly aggressive. Perhaps we could scope it to impacted actions?